src/Filters/Block.php

```diff
@@ -0,0 +1,1029 @@
+                    'nbr' => $nbProducts
+                ]
+            ];
+        }
+
+        $this->restorePriceAndWeightFilters(
+            $this->facetedSearchAdapter->getInitialPopulation(),
+            $priceMinFilter,
+            $priceMaxFilter,
+            $weightFilter
+        );
+
+        return $priceBlock;
+    }
+
+    /**
+     * Price / weight filter block should not apply there own filters
+     * otherwise they will always disappear if we filter on price / weight
+     * because only one choice will remain
+     *
+     * @param InterfaceAdapter $filteredSearchAdapter
+     *
+     * @return array
+     */
+    private function ignorePriceAndWeightFilters(InterfaceAdapter $filteredSearchAdapter)
+    {
+        // disable the current price and weight filters to compute ranges
+        $priceMinFilter = $filteredSearchAdapter->getFilter('price_min');
+        $priceMaxFilter = $filteredSearchAdapter->getFilter('price_max');
+        $weightFilter = $filteredSearchAdapter->getFilter('weight');
+        $filteredSearchAdapter->resetFilter('price_min');
+        $filteredSearchAdapter->resetFilter('price_max');
+        $filteredSearchAdapter->resetFilter('weight');
+
+        return [
+            $priceMinFilter,
+            $priceMaxFilter,
+            $weightFilter,
+        ];
+    }
+
+    /**
+     * Restore price and weight filters
+     *
+     * @param InterfaceAdapter $filteredSearchAdapter
+     * @param integer $priceMinFilter
+     * @param integer $priceMaxFilter
+     * @param integer $weightFilter
+     *
+     * @return array
+     */
+    private function restorePriceAndWeightFilters(
+        $filteredSearchAdapter,
+        $priceMinFilter,
+        $priceMaxFilter,
+        $weightFilter
+    ) {
+        // put back the price and weight filters
+        $filteredSearchAdapter->setFilter('price_min', $priceMinFilter);
+        $filteredSearchAdapter->setFilter('price_max', $priceMaxFilter);
+        $filteredSearchAdapter->setFilter('weight', $weightFilter);
+    }
+
+    /**
+     * Get the weight filter block
+     *
+     * @param array $filter
+     * @param array $selectedFilters
+     * @param int $nbProducts
+     *
+     * @return array
+     */
+    private function getWeightRangeBlock($filter, $selectedFilters, $nbProducts)
+    {
+        $weightBlock = [
+            'type_lite' => 'weight',
+            'type' => 'weight',
+            'id_key' => 0,
+            'name' => Context::getContext()->getTranslator()->trans('Weight', [], 'Modules.Facetedsearch.Shop'),
+            'max' => '0',
+            'min' => null,
+            'unit' => Configuration::get('PS_WEIGHT_UNIT'),
+            'specifications' => [],
+            'filter_show_limit' => $filter['filter_show_limit'],
+            'filter_type' => Converter::WIDGET_TYPE_SLIDER,
+            'list_of_values' => [],
+        ];
+
+        $filteredSearchAdapter = $this->facetedSearchAdapter->getFilteredSearchAdapter('weight');
```

Comment:
To don't reset slider ranges filters, otherwise:
- You have a slider with `10kg - 100kg`
- You select `50kg - 100kg`
Without this you'll have a new slider with `50kg - 100kg` instead of being able to return to `10kg - 100kg`

Reply:
Ok, I see the idea but I'm not sure it is used this way here.
In `getPriceRangeBlock` it makes sens because after the call to `ignorePriceAndWeightFilters` min/max are set like this:

```php
list($priceBlock['min'], $priceBlock['max']) = $this->facetedSearchAdapter->getInitialPopulation()->getMinMaxPriceValue();
```

But in that case `getInitialPopulation` is not used, only `$filteredSearchAdapter` so either this can be simplifies by removing `ignorePriceAndWeightFilters / restorePriceAndWeightFilters` or the min/max should be fetch from `initialPopulation`

Reply:
Nope, initialPopulation is used inside `getFilteredSearchAdapter`.

Reply:
Yes but `getFilteredSearchAdapter` is called before `ignorePriceAndWeightFilters` so no influence in that case.

Reply:
As i know, it because both wieght and price must be reset, in case you use weight and price filters. I'll investigate on it ;)

Reply:
You were right! I remove this part of code :D

Reply:
I spoke too fast, there is something weird now :P

Reply:
:D that would have surprised me!

Reply:
So, I can't remove the ignorePriceAndWeightFilters because it'll remove price if there is no price available, but we wan to keep it displayed all the time.
I also fix something with the price filter.